pygmt/params/position.py

```diff
@@ -0,0 +1,209 @@
+    - ``type="outside"``: Mirror opposite of the *reference point* justification code
+    - Other types: ``"MC"`` (middle center) for map rose and scale, ``"BL"``
+      (bottom-left) for other embellishments
+
+    **Offset**
+
+    The ``offset`` parameter shifts the *anchor point* from its default position.
+    Offsets are applied to the projected plot coordinates, with positive values moving
+    in the direction indicated by the *anchor point*'s justification code. It should be
+    a single value (applied to both x and y) or as (*offset_x*, *offset_y*).
+
+    Examples
+    --------
+    Position the GMT logo at map coordinates (3, 3) with the logo's middle-left point as
+    the anchor, offset by (0.2, 0.2):
+
+    >>> import pygmt
+    >>> from pygmt.params import Position
+    >>> fig = pygmt.Figure()
+    >>> fig.basemap(region=[0, 10, 0, 10], projection="X10c", frame=True)
+    >>> fig.logo(
+    ...     position=Position((3, 3), type="mapcoords", anchor="ML", offset=(0.2, 0.2)),
+    ...     box=True,
+    ... )
+    >>> fig.show()
+
+    Position the GMT logo at the top-left corner inside the plot:
+
+    >>> fig = pygmt.Figure()
+    >>> fig.basemap(region=[0, 10, 0, 10], projection="X10c", frame=True)
+    >>> fig.logo(position=Position("TL", type="inside", offset="0.2c"), box=True)
+    >>> fig.show()
+    """
+
+    #: Location of the reference point on the plot. The format depends on ``type``:
+    #:
+    #: - ``type="mapcoords"``: (*longitude*, *latitude*)
+    #: - ``type="plotcoords"``: (*x*, *y*) with plot units
+    #: - ``type="boxcoords"``: (*nx*, *ny*)
+    #: - ``type="inside"`` or ``"outside"``:
+    #:   :doc:`2-character justification codes </techref/justification_codes>`
+    location: Sequence[float | str] | AnchorCode
+
+    #: Types of the reference point. Valid values are:
+    #:
+    #: - ``"mapcoords"``: Map/Data coordinates
+    #: - ``"plotcoords"``: Plot coordinates
+    #: - ``"boxcoords"``: Normalized coordinates
+    #: - ``"inside"`` or ``"outside"``: Justification codes
+    #:
+    #: If not specified, defaults to ``"inside"`` if ``location`` is a justification
+    #: code; otherwise defaults to ``"plotcoords"``.
+    type: (
+        Literal["mapcoords", "inside", "outside", "boxcoords", "plotcoords"] | None
+    ) = None
+
+    #: Anchor point on the embellishment using a
+    #: :doc:`2-character justification codes </techref/justification_codes>`.
+    #: If ``None``, defaults are applied based on ``type`` (see above).
+    anchor: AnchorCode | None = None
+
+    #: Offset for the anchor point as a single value or (*offset_x*, *offset_y*).
+    #: If a single value is given, the offset is applied to both x and y directions.
+    offset: Sequence[float | str] | None = None
+
+    def _validate(self):
+        """
+        Validate the parameters.
+        """
+        _valid_anchors = {f"{h}{v}" for v in "TMB" for h in "LCR"} | {
+            f"{v}{h}" for v in "TMB" for h in "LCR"
+        }
```

Comment:
We probably can reuse these codes to validate anchor codes in other wrappers, e.g., the `justify` parameter in the `text` module. But I feel we can do it in the future.